core/src/autogluon/core/utils/feature_selection.py

```diff
@@ -0,0 +1,537 @@
+    """
+    Return a list of features sorted by feature importance calculation priority in ascending order.
+    If prev_importance_df does not exist and not using auto_threshold, return the original list.
+    If prev_importance_df exists, features whose importance scores have never been calculated are
+    prioritized first followed by features whose importance scores are from previous fitted models and
+    lastly the rest of the features sorted by previous importance scores estimates in ascending order. If using
+    auto_threshold mode, noise columns are prioritized first since their scores are needed to determine
+    the pruning threshold.
+    """
+    is_first_run = prev_importance_df is None
+    auto_threshold = len(prioritized) > 0
+    if not is_first_run:
+        prev_deleted_features = [feature for feature in prev_importance_df.index if feature not in features]
+        prev_importance_df = prev_importance_df.drop(prev_deleted_features)
+        unevaluated_rows = prev_importance_df[prev_importance_df['importance'].isna()]
+        prev_fit_evaluated_rows = prev_importance_df[~(prev_importance_df['importance'].isna()) &
+                                                      (prev_importance_df.index.isin(using_prev_fit_fi))].sort_values(by='importance')
+        curr_fit_evaluated_rows = prev_importance_df[~(prev_importance_df['importance'].isna()) &
+                                                     ~(prev_importance_df.index.isin(using_prev_fit_fi))].sort_values(by='importance')
+        features = unevaluated_rows.index.tolist() + prev_fit_evaluated_rows.index.tolist() + curr_fit_evaluated_rows.index.tolist()
+    if auto_threshold:
+        features = list(prioritized) + [feature for feature in features if feature not in prioritized]
+    return features
+
+
+class FeatureSelector:
+    def __init__(self, model: AbstractModel, time_limit: float, seed: int = 0, keep_models: bool = False, raise_exception=False) -> None:
+        # TODO: Make this work with unlabelled data
+        assert time_limit is not None, "Time limit cannot be unspecified."
+        assert model.is_valid(), "Model must have been fit."
+        self.original_model = model
+        self.base_model = model.convert_to_template()
+        self.base_model.rename(f'FeatureSelector_{self.base_model.name}')
+        self.original_time_limit = time_limit
+        self.time_limit = time_limit
+        self.rng = np.random.default_rng(seed)
+        self.keep_models = keep_models
+        self.is_bagged = isinstance(model, BaggedEnsembleModel)
+        self.model_fit_time = model.fit_time
+        self.model_predict_time = model.predict_time
+        self.original_val_score = model.val_score
+        self.trained_models = []
+        self.importance_dfs = []
+        self.attempted_removals = set()
+        self.noise_prefix = 'AG_normal_noise'
+        self.safety_time_multiplier = 1.1
+        self.max_n_shuffle = 100
+        self.max_time_budget_fi = 300
```

Comment:
seems like an odd limitation. What if 24 hours of time is given? Surely we could spare more than 5 minutes on FI calculations.